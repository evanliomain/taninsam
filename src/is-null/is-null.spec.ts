import { describe, expect, test } from 'vitest';
import { isNull } from './is-null';

describe('isNull function', () => {
  const input = [1, 2, 3, 4, 5];

  test('is a pure function', () => {
    isNull(input);
    expect(input).toMatchSnapshot();
  });

  test('null |> isNull === true', () => {
    expect(isNull(null)).toBeTruthy();
  });

  test('undefined |> isNull === false', () => {
    expect(isNull(undefined)).toBeFalsy();
  });

  test('[] |> isNull === false', () => {
    expect(isNull([])).toBeFalsy();
  });

  test('0 |> isNull === false', () => {
    expect(isNull(0)).toBeFalsy();
  });

  test('"" |> isNull === false', () => {
    expect(isNull('')).toBeFalsy();
  });
});
