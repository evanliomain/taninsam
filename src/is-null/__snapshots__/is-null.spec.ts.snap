// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`isNull function > is a pure function 1`] = `
[
  1,
  2,
  3,
  4,
  5,
]
`;
